include/can/core/freertos_can_dispatch.hpp

```diff
@@ -0,0 +1,91 @@
+#pragma once
+
+#include "FreeRTOS.h"
+#include "can_message_buffer.hpp"
+#include "common/core/freertos_message_buffer.hpp"
+#include "dispatch.hpp"
+#include "message_core.hpp"
+
+using namespace message_core;
+using namespace message_buffer;
+using namespace can_message_buffer;
+using namespace can_dispatch;
+using namespace freertos_message_buffer;
+
+namespace freertos_can_dispatch {
+
+/**
+ * A FreeRTOS task entry point that polls a MessageBuffer.
+ * @tparam BufferType The MessageBuffer type
+ * @tparam Listener The CanMessageBufferListener type
+ */
+template <MessageBuffer BufferType, CanMessageBufferListener Listener>
+class FreeRTOSCanBufferPoller {
+  public:
+    /**
+     * Constructor
+     * @param buffer The MessageBuffer instance (ie FreeRTOSMessageBuffer)
+     * @param listener The listener to be called back when a new message is in
+     * the buffer.
+     */
+    FreeRTOSCanBufferPoller(BufferType& buffer, Listener& listener)
+        : buffer{buffer}, listener{listener}, reader{buffer, listener} {}
+
+    /**
+     * Task entry point.
+     */
+    void operator()() {
```

Comment:
is this stoppable by calling some method on `reader`? if not we should probably annotate it [`[[noreturn]]`](https://en.cppreference.com/w/cpp/language/attributes/noreturn)